packages/react-instantsearch/src/widgets/Chat.tsx

```diff
@@ -147,34 +166,64 @@ export function Chat<TObject extends RecordWithObjectID>({
     },
   });
 
+  const handleClear = React.useCallback(() => {
+    if (!messages || messages.length === 0) return;
+    setIsClearing(true);
+  }, [messages]);
+
+  const handleClearTransitionEnd = React.useCallback(() => {
+    setMessages([]);
+    clearError();
+    setIsClearing(false);
+  }, [setMessages, clearError]);
+
   return (
     <ChatUiComponent
       open={open}
+      maximized={maximized}
       toggleButtonProps={{
         open,
         onClick: () => setOpen(!open),
         ...toggleButtonProps,
       }}
+      headerProps={{
+        onClose: () => setOpen(false),
+        maximized,
+        onToggleMaximize: () => setMaximized(!maximized),
+        onClear: handleClear,
+        canClear: messages && messages.length > 0 && !isClearing,
+        ...headerProps,
+      }}
       messagesProps={{
+        status,
+        onReload: (messageId) => regenerate({ messageId }),
         messages,
         tools,
         indexUiState,
         setIndexUiState,
+        isClearing,
+        onClearTransitionEnd: handleClearTransitionEnd,
+        scrollRef: scrollRef as unknown as MutableRef<HTMLDivElement>,
+        contentRef: contentRef as unknown as MutableRef<HTMLDivElement>,
```

Comment:
is this because react and ui components disagree what a mutable ref is? I guess it's only details but the main api is the same?

Reply:
I avoided using `unknown` by doing the following:
```ts
scrollRef: { current: scrollRef.current as HTMLDivElement },
contentRef: { current: contentRef.current as HTMLDivElement },
 ```